tpm2/keydata_v0.go

```diff
@@ -173,7 +173,7 @@ func (d *keyData_v0) ValidateData(tpm *tpm2.TPMContext, session tpm2.SessionCont
 		return nil, keyDataError{errors.New("cannot determine if PCR policy counter has a valid authorization policy: algorithm unavailable")}
 	}
 	pcrPolicyCounterAuthPolicies := d.PolicyData.StaticData.PCRPolicyCounterAuthPolicies
-	expectedPCRPolicyCounterAuthPolicies := computeV0PinNVIndexPostInitAuthPolicies(pcrPolicyCounterPub.NameAlg, authKeyName)
+	expectedPCRPolicyCounterAuthPolicies := computeV0PinNVIndexPostInitAuthPolicies(pcrPolicyCounterPub.NameAlg, authPublicKey.Name())
```

Comment:
`ComputeName` is the same as the previous `Name` - it would return an error if the name could not be computed, eg, because the algorithm isn't available or the object doesn't serialize. The new `Name` drops the error, but returns an invalid name where an error would have been reported, which can be detected wherever the name is used. The intention of this is so that the various representations of resources that have names in go-tpm2 can all implement the same interface (https://pkg.go.dev/github.com/canonical/go-tpm2@master/policyutil#Named and https://pkg.go.dev/github.com/canonical/go-tpm2@master/objectutil#Named), where they were different before (some returned an error, some didn't).

The calls to `Name` in this PR are either on objects that we've just created (so we know are good) or are on objects that have just been validated.

I've pushed a small update that adds some notes to functions to clarify where they expect arguments to be valid.